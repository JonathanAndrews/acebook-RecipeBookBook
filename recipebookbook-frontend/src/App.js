import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import PostsContainer from './components/PostsContainer';


class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">RecipeBookBook!</h1>
        </header>
        <PostsContainer />
      </div>
    );
  }
}

export default App;
