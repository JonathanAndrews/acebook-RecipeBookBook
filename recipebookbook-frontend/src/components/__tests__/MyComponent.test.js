
import React from 'react';
import { shallow } from 'enzyme';
import  App from '../../App.js';
import axios from 'axios';
import PostContainer from '../PostsContainer.js'

describe('should render title', () => {

  it('should render correctly with no props', () => {
    const component = shallow(<App/>);
    expect(component).toMatchSnapshot();
  });
  

  it('should have title', () => {
    const component = shallow(<App/>);
    var node = component.find('h1')
    expect(node.hasClass('App-title')).toBeTruthy();
    expect(node.html()).toEqual('<h1 class="App-title">RecipeBookBook!</h1>')
  });
})

describe('should render posts', () => {
  it('should fetch a post', () => {
    const getSpy = jest.spyOn(axios, 'get');
    const postInstance = shallow(
      <PostContainer/>
    );
    expect(getSpy).toBeCalled();
  })
})
