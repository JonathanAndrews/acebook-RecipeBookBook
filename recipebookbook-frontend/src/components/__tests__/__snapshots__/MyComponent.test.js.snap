// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render title should render correctly with no props 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      RecipeBookBook!
    </h1>
  </header>
  <PostsContainer />
</div>
`;
